src/Cms/FileActions.php

```diff
@@ -288,11 +288,10 @@ public function delete(): bool
 			// remove all public versions and clear the UUID cache
 			$file->unpublish();
 
-			// delete all changes first
-			$file->version('changes')->delete('*');
-
-			// delete all latest versions as last step.
-			$file->version('latest')->delete('*');
+			// delete all versions
+			foreach ($file->versions() as $version) {
+				$version->delete('*');
+			}
```

Comment:
These changes in the delete actions make the version order implicit. Nothing would prevent us from breaking this order by changing it in `Versions::load()` by accident for some other feature or fix.

If the order is indeed required as stated in the previous code comments, here are some ideas:

- Add a param for `$model->versions()` that controls the order
- Add a code comment to `Versions::load()` and an unit test that both ensure that `changes` is always first, pointing to the `delete` actions that need it

Reply:
I think I would go for a comment in the versions class. I don't think it's something specific of the delete method that would be better dealt with a parameter. It's always the case that the latest version is the umbrella one so to say and should go last. Likely also in a future where there might be revisions too.

Reply:
So far, the unit tests should also protect us, right?

Reply:
Yes. I guess adding a comment both on the method and in the unit test to explain that the order is important should be sufficient.

Reply:
☑️